cpp/src/arrow/compute/kernels/minmax.cc

```diff
@@ -84,6 +88,8 @@ class MinMaxAggregateFunction final
   Status Consume(const Array& array, StateType* state) const override {
     StateType local;
 
+    local.has_nulls = array.null_count() > 0;
```

Comment:
There's no need to visit the array's values if we know that the result will be null:
```suggestion
    local.has_nulls = array.null_count() > 0;
    if (local.has_nulls && options_.null_handling == MinMaxOptions::OUTPUT_NULL) {
      return Status::OK();
    }
```

Reply:
@bkietz just saw this notification
fixed now. thanks for the careful review!